pkg/envfuncs/kwok_funcs.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envfuncs
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/support/kwok"
+)
+
+type kwokContextKey string
+
+// GetKwokClusterFromContext helps extract the kwok.Cluster object from the context.
+func GetKwokClusterFromContext(ctx context.Context, clusterName string) (*kwok.Cluster, bool) {
+	kwokCluster := ctx.Value(kwokContextKey(clusterName))
+	if kwokCluster == nil {
+		return nil, false
+	}
+	cluster, ok := kwokCluster.(*kwok.Cluster)
+	return cluster, ok
+}
+
+// CreateKwokCluster returns an env.Func that is used to
+// create a kwok cluster that is then injected in the context
+// using the name as a key.
+//
+// NOTE: the returned function will update its env config with the
+// kubeconfig file for the config client.
+func CreateKwokCluster(clusterName string) env.Func {
+	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		k := kwok.NewCluster(clusterName)
+		kubecfg, err := k.Create()
+		if err != nil {
+			return ctx, err
+		}
+
+		// update envconfig  with kubeconfig
+		cfg.WithKubeconfigFile(kubecfg)
+
+		// stall, wait for pods initializations
+		if err := waitForKubeNamespaces(cfg.Client()); err != nil {
```

Comment:
It is best to use `--wait` to wait for ready.

Reply:
thanks. I changed it to use `--wait`